GA4 copy changes to front end

nick-mon1 opened the pull request:
## Summary

Covers javascript and template (`charts.html`) updates to address the migration from `GA3` to `GA4`.


## Work to Do

- [x] javascript impact
- [x] setup and hit `fake-data/live` reports - see [README.md](https://github.com/18F/analytics.usa.gov#developing-with-local-data)
	- [x] fix 404
- [x] replace `fake-data/live` with `ga4-data` reports to see where the code fails
- ~[ ] address code updates to handle the new `GA4` reports~



## Running Locally

1. `bundle install`
2. `npm install`
3. `make dev` will run local server with development configuration
4. `npm run start` runs `make dev` and watches style changes.

## Template Blocks

**15 data block sections displaying d3 chart data**

- realtime
- today
- devices
- browsers
	- ie (hidden)
- os
	- windows (hidden)
- cities
- countries
- us and territories
- international
- top pages realtime
- top pages 7 days
- top pages 30 days
- top downloads

**d3 chart types**

- bar chart
- bar chart basic
- bar chart with label
- time series


## Javascript Impact

Some values displayed in the charts are generated off the reports with further calculations done in the javascript.


| js                     | function                                                                                                        | impact                                                                                               |
| ---------------------- | --------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------- |
| js/index.js            | includes the main dom selection and rendering queue of components, and the entry point for the webpack bundler. | moderate impact potentially with os, browsers, countries bar charts                                  |
| js/lib/barchart.js     | the d3 configuration of the bar charts                                                                          | zero to minimal impact                                                                               |
| js/lib/blocks.js       | an object of the specific components                                                                            | moderate to high impact for 11 code block sections                                                   |
| js/lib/consoleprint.js | the console messages displayed to users                                                                         | no impact                                                                                            |
| js/lib/eventhandler.js | handles certain dom events                                                                                      | no impact                                                                                            |
| js/lib/exceptions.js   | agency data to be changed by discrete exception rules                                                           | minimal impact, should consult if we need to update any exceptions                                   |
| js/lib/formatters.js   | methods to help format the display of visualization scales and values                                           | minimal to high impact, will need to check how GA4 data has changed and update functions accordingly |
| js/lib/renderblock.js  | d3 manipulator to load and render data for a component block, executes on 15 data blocks                        | low to moderate impact, handles block lifecycle and events, not data related                         |
| js/lib/territories.js  | handles which territories are displayed in the countries bar chart                                              | no impact to minimal impact                                                                          |
| js/lib/timeseries.js   | the d3 configuration of the timeseries charts                                                                   | no impact to moderate impact depending on GA4 changes                                                |
| js/lib/transformers.js | helper methods to manipulate and consolidate raw data into proportional data.                                   | no impact to moderate impact depending on GA4 changes                                                |



## Reports

According to the README.md, running the jekyll app locally can reference test data located `./fake-data/live`.


- [ ] document differences between GA3 and GA4 files?
	- [ ] have fields been moved to other files?
	- [ ] new fields that contain same data as old fields?
	- [ ] what fields are not supported anymore?
- [ ] how to handle data for a block that is split in several reports?
	- [ ] could create a new report that merges data to work with the data-blocks
- [ ] update charts.html with new report filenames


**blank fields mean there is no matching report**

| GA4 (./ga4-data)                | GA3 (/fake-data/live)              |
| -------------------------------- | ---------------------------- |
| browsers.csv                     | browsers.csv                 |
| browsers.json                    | browsers.json                |
| device_model.csv                 |                              |
| device_model.json                |                              |
| devices.csv                      | devices.csv                  |
| devices.json                     | devices.json                 |
| ie.csv                           | ie.csv                       |
| ie.json                          | ie.json                      |
| language.csv                     |                              |
| language.json                    |                              |
| last-48-hours.csv                |                              |
| last-48-hours.json               |                              |
| os-browsers.csv                  |                              |
| os-browsers.json                 |                              |
| os.csv                           | os.csv                       |
| os.json                          | os.json                      |
| realtime.csv                     | realtime.json                |
| realtime.json                    |                              |
| screen-size.csv                  |                              |
| screen-size.json                 |                              |
| today.csv                        |                              |
| today.json                       | today.json                   |
|                                  | all-domains-30-days.csv      |
|                                  | all-pages-realtime.csv       |
| top-domains-30-days.csv          | top-domains-30-days.csv      |
| top-domains-30-days.json         | top-domains-30-days.json     |
| top-domains-7-days.csv           |                              |
| top-domains-7-days.json          | top-domains-7-days.json      |
| top-landing-pages-30-days.csv    |                              |
| top-landing-pages-30-days.json   |                              |
| top-pages-30-days.csv            |                              |
| top-pages-30-days.json           |                              |
| top-pages-7-days.csv             |                              |
| top-pages-7-days.json            |                              |
| top-pages-realtime.csv           |                              |
| top-pages-realtime.json          | top-pages-realtime.json      |
| top-traffic-sources-30-days.csv  |                              |
| top-traffic-sources-30-days.json |                              |
|                                  | top-cities-realtime.json     |
|                                  | top-countries-realtime.json  |
|                                  | top-downloads-7-days.csv     |
|                                  | top-downloads-7-days.json    |
|                                  | top-downloads-yesterday.json |
| users.csv                        |                              |
| users.json                       |                              |
| windows-browsers.csv             |                              |
| windows-browsers.json            |                              |
| windows-ie.csv                   |                              |
| windows-ie.json                  |                              |
| windows.csv                      | windows.csv                  |
| windows.json                     | windows.json                 |





## This pull request is ready to merge when...
- [ ] Tests have been updated (and all tests are passing)
- [ ] This code has been reviewed by someone other than the original author
- [ ] The change has been documented

ryanwoldatwork commented on _includes/charts.html:
```diff
@@ -21,6 +21,9 @@
       </section>
       -->
 <div class="grid-row">
+  <h3>site.data_url: {{ site.data_url }}</h3>
```
is this for debugging?

nick-mon1 replied:
Yes, it's now removed

ryanwoldatwork commented on sass/_uswds-theme-custom-styles.scss:
```diff
@@ -140,22 +136,22 @@ ul li:before {
   padding-right: 0;
 }
 
+// TODO: replace 13 instances
```
👀

ryanwoldatwork reviewed (approved):
generally, looking good. 

a few small things

mejiaj commented:
Now that data is working I've done two things:

1. Separate chart markup so its easier to maintain [0580c90].
    - There's a new section `_includes/charts` that has all of the chart blocks.
2. Refactor/add styling to make homepage match live.

**Current** (develop)
<img width="1000" alt="image" src="https://github.com/18F/analytics.usa.gov/assets/3385219/9a41df5f-ab63-4aae-8a36-a00693493607">


**After**
<img width="1000" alt="image" src="https://github.com/18F/analytics.usa.gov/assets/3385219/ec2f9b50-d773-44a1-a303-8a9bde8d8e7d">

nick-mon1 commented:
Updated various copy changes for GA4.

- [x] 12m should read 12M
- [x] 1am 2am should read 1AM 2AM
- [x] where is the 90 (now 30) days is set in blocks.js:73 (#total_visitors)
- [x] display current visitors count
- [x] update browser to web browsers in the sessions section
- [x] remove hyper links from first paragraph last line
- [x] identifier change project to product, remove cloud.gov line
- [x] removed "less than 5%" copy
- [x] removed warning notice
- [x] removed debugging output
- [x] update "to hostnames" to "on hostnames"
